lib/actions/preload.coffee

```diff
@@ -0,0 +1,206 @@
+		name: "#{build.push_timestamp} - #{build.commit_hash}"
+		value: build.commit_hash
+	return form.ask
+		message: 'Select a build'
+		type: 'list'
+		default: LATEST
+		choices: choices
+
+offerToDisableAutomaticUpdates = (Promise, form, resin, application, commit) ->
+	if commit == LATEST or not application.should_track_latest_release
+		return Promise.resolve()
+	message = '''
+
+		This application is set to automatically update all devices to the latest available version.
+		This might be unexpected behaviour: with this enabled, the preloaded device will still
+		download and install the latest build once it is online.
+
+		Do you want to disable automatic updates for this application?
+	'''
+	form.ask
+		message: message,
+		type: 'confirm'
+	.then (update) ->
+		if not update
+			return
+		resin.pine.patch
+			resource: 'application'
+			id: application.id
+			body:
+				should_track_latest_release: false
+
+module.exports =
+	signature: 'preload <image>'
+	description: '(beta) preload an app on a disk image'
+	help: '''
+		Warning: "resin preload" requires Docker to be correctly installed in
+		your shell environment. For more information (including Windows support)
+		please check the README here: https://github.com/resin-io/resin-cli .
+
+		Use this command to preload an application to a local disk image with a
+		built commit from Resin.io.
+		This can be used with cloud builds, or images deployed with resin deploy.
+
+		Examples:
+		  $ resin preload resin.img --app 1234 --commit e1f2592fc6ee949e68756d4f4a48e49bff8d72a0 --splash-image some-image.png
+		  $ resin preload resin.img
+	'''
+	permission: 'user'
+	primary: true
+	options: dockerUtils.appendConnectionOptions [
+		{
+			signature: 'app'
+			parameter: 'appId'
+			description: 'id of the application to preload'
+			alias: 'a'
+		}
+		{
+			signature: 'commit'
+			parameter: 'hash'
+			description: 'a specific application commit to preload (ignored if no appId is given)'
+			alias: 'c'
+		}
+		{
+			signature: 'splash-image'
+			parameter: 'splashImage.png'
+			description: 'path to a png image to replace the splash screen'
+			alias: 's'
+		}
+		{
+			signature: 'dont-detect-flasher-type-images'
+			boolean: true
+			description: 'Disables the flasher type images detection: treats all images as non flasher types'
+		}
+	]
+	action: (params, options, done) ->
+		_ = require('lodash')
+		Promise = require('bluebird')
+		resin = require('resin-sdk-preconfigured')
+		streamToPromise = require('stream-to-promise')
+		form = require('resin-cli-form')
+		preload = require('resin-preload')
+		errors = require('resin-errors')
+		{ expectedError } = require('../utils/patterns')
+
+		options.image = params.image
+		options.appId = options.app
+		delete options.app
+
+		options.dontDetectFlasherTypeImages = options['dont-detect-flasher-type-images']
+		delete options['dont-detect-flasher-type-images']
+
+		dockerUtils.getDocker(options)
+		.then (docker) ->
```

Comment:
Could you add maybe a little vertical spacing to break up the logic a bit? I imagine I'm not the only one that struggles to parse walls of code.